sweep/fee_bumper.go

```diff
@@ -0,0 +1,79 @@
+package sweep
+
+import (
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+)
+
+type FeeBumpStrategy uint8
+
+const (
+	// FeeBumpStrategyAggressive indicates that the fee bumping should
+	// always use the larger fee rate between the calculated fee rate and
+	// the estimated fee rate.
+	FeeBumpStrategyAggressive FeeBumpStrategy = iota
+
+	// FeeBumpStrategyConservative indicates that the fee bumping should
+	// always use the smaller fee rate between the calculated fee rate and
+	// the estimated fee rate.
+	FeeBumpStrategyConservative
+
+	// TODO(yy): define more sophisticated strategies.
+)
+
+// Bumper defines an interface that can be used by other subsystems for fee
+// bumping.
+type Bumper interface {
+	// Broadcast is used to publish the tx created from the given inputs
+	// specified in the request. It handles the tx creation, broadcasts it,
+	// and monitors its confirmation status for potential fee bumping. It
+	// returns a chan that the caller can use to receive updates about the
+	// broadcast result and potential RBF attempts.
+	//
+	// NOTE: it's expected to have a BumpResult sent immediately once the
+	// first broadcast attempt is finished.
+	Broadcast(req *BumpRequest) (<-chan *BumpResult, error)
+}
+
+// BumpRequest is used by the caller to give the Bumper the necessary info to
+// create and manage potential fee bumps for a set of inputs.
+type BumpRequest struct {
+	// Inputs is the set of inputs to sweep.
+	Inputs InputSet
+
+	// ChangeScript is the script to send the change output to.
+	ChangeScript []byte
```

Comment:
FWIW that terminology is used mainly for the co-op close process.

Reply:
That's fine, but I generally think we shouldn't have []byte anywhere except JIT before deserialization or JIT after serialization. We should have semantic domain types everywhere. Since we already have a DeliveryAddress, that seems appropriate, if we need to relax the terminology to Address, or ScriptPubKey that's fine too.